tls/test_tls_handshake.py

```diff
@@ -239,22 +267,115 @@ def ADDED_CHANGECIPHERSPEC(self):
+                self.add_msg(
+                    TLSApplicationData(data=f"GET / HTTP/1.1\r\nHost: tempesta-tech.com\r\n\r\n")
+                )
                 self.add_record()
                 self.add_msg(TLSFinished())
                 raise self.ADDED_CLIENTFINISHED()
 
         conn = TlsHandshake()
         res = conn.do_12(automaton=_ModifiedTLSClientAutomaton)
         self.assertFalse(res, "Bad handshake successfully processed")
-        
-        
+
     def test_old_handshakes(self):
         self.start_all()
         res = TlsHandshakeStandard().do_old()
         self.assertTrue(res, "Wrong old handshake result: %s" % res)
 
 
+class TlsHandshakeDheRsaTest(tester.TempestaTest):
+    clients = [
+        {
+            "id": "tls-perf-DHE-RSA-AES128-GCM-SHA256",
+            "type": "external",
+            "binary": "tls-perf",
+            "cmd_args": (
+                "-c DHE-RSA-AES128-GCM-SHA256 -C prime256v1 -l 1000 -t 2 -T %s ${server_ip} 443"
+                % (tf_cfg.cfg.get("General", "Duration"))
+            ),
+        },
+        {
+            "id": "tls-perf-DHE-RSA-AES256-GCM-SHA384",
+            "type": "external",
+            "binary": "tls-perf",
+            "cmd_args": (
+                "-c DHE-RSA-AES256-GCM-SHA384 -C prime256v1 -l 1000 -t 2 -T %s ${server_ip} 443"
+                % (tf_cfg.cfg.get("General", "Duration"))
+            ),
+        },
+        {
+            "id": "tls-perf-DHE-RSA-AES128-GCM",
+            "type": "external",
+            "binary": "tls-perf",
+            "cmd_args": (
+                "-c DHE-RSA-AES128-GCM -C prime256v1 -l 1000 -t 2 -T %s ${server_ip} 443"
+                % (tf_cfg.cfg.get("General", "Duration"))
+            ),
+        },
+        {
+            "id": "tls-perf-DHE-RSA-AES256-GCM",
+            "type": "external",
+            "binary": "tls-perf",
+            "cmd_args": (
+                "-c DHE-RSA-AES256-GCM -C prime256v1 -l 1000 -t 2 -T %s ${server_ip} 443"
+                % (tf_cfg.cfg.get("General", "Duration"))
+            ),
+        },
+    ]
+
+    tempesta_tmpl = """
+        cache 0;
+        listen 443 proto=https;
+
+        tls_certificate %s;
+        tls_certificate_key %s;
+        tls_match_any_server_name;
+
+        srv_group srv_grp1 {
+            server ${server_ip}:8000;
+        }
+        vhost tempesta-tech.com {
+            proxy_pass srv_grp1;
+        }
+        http_chain {
+            host == "tempesta-tech.com" -> tempesta-tech.com;
+            -> block;
+        }
+    """
+
+    def setUp(self):
+        self.cgen = CertGenerator()
+        self.cgen.key = {"alg": "rsa", "len": 4096}
+        self.cgen.sign_alg = "sha256"
+        self.cgen.generate()
+        self.tempesta = {
+            "config": self.tempesta_tmpl % self.cgen.get_file_paths(),
+            "custom_cert": True,
+        }
+        tester.TempestaTest.setUp(self)
+
+    def basicTest(self, alg):
```

Comment:
PEP 8 requires using lowercase for function name.

Reply:
Fixed